packages/plugin-ext/src/plugin/known-commands.ts

```diff
@@ -17,9 +17,9 @@
 import { Range as R, Position as P, Location as L } from 'vscode-languageserver-types';
 import URI from 'vscode-uri';
 import * as theia from '@theia/plugin';
-import { cloneAndChange } from './objects';
-import { Position, Range, Location } from '../plugin/types-impl';
-import { fromPosition, fromRange, fromLocation } from '../plugin/type-converters';
+import { cloneAndChange } from '../common/objects';
```

Comment:
Could someone elaborate except moving this file and renaming things what was fixes in this file? It is used for all commands triggered by the plugin host process would be bad to break it.

Reply:
For `vscode.executeReferenceProvider` and `vscode.executeImplementationProvider`, as they return location array as result,  it is required to convert the result from `model.*` objects to `theia.*` objects in order to be accepted by VSCode Reference View plugin.
So, optional property is added to the mapping to specify the converter function that is to be used to convert the results. As it is optional, there is no need to add this property to all the other existing mappings. Currently, `CONVERT_MONACO_TO_VSCODE` is added to convert the results array into an array of `theia.Location` objects. It's used to convert the results of `vscode.executeReferenceProvider` and `vscode.executeImplementationProvider` command execution.
The only change for the `old` mappings is that `MONACO_CONVERSION_IDENTITY` function used for arguments conversion is renamed to `CONVERT_VSCODE_TO_MONACO` (in order to indicate the fact that it really changes arguments -  requested by @tsmaeder )

Reply:
Unfortunately, the github diff algorithm is being really stupid here: there are loads of single line changes in the command definitions, but GH picks it up as a two entirely changed blocks of code 🤷‍♂